src/textual/_worker_manager.py

```diff
@@ -0,0 +1,164 @@
+
+
+@rich.repr.auto(angular=True)
+class WorkerManager:
+    """An object to manager a number of workers.
+
+    You will not have to construct this class manually, as widgets, screens, and apps
+    have a worker manager accessibly via a `workers` attribute.
+
+    """
+
+    def __init__(self, app: App) -> None:
+        """Initialize a worker manager.
+
+        Args:
+            app: An App instance.
+        """
+        self._app = app
+        """A reference to the app."""
+        self._workers: set[Worker] = set()
+        """The workers being managed."""
+
+    def __rich_repr__(self) -> rich.repr.Result:
+        counter: Counter[WorkerState] = Counter()
+        counter.update(worker.state for worker in self._workers)
+        for state, count in sorted(counter.items()):
+            yield state.name, count
+
+    def __iter__(self) -> Iterator[Worker[Any]]:
+        return iter(sorted(self._workers, key=attrgetter("_created_time")))
+
+    def __reversed__(self) -> Iterator[Worker[Any]]:
+        return iter(
+            sorted(self._workers, key=attrgetter("_created_time"), reverse=True)
+        )
+
+    def __bool__(self) -> bool:
+        return bool(self._workers)
+
+    def __len__(self) -> int:
+        return len(self._workers)
+
+    def __contains__(self, worker: object) -> bool:
+        return worker in self._workers
+
+    def add_worker(
+        self, worker: Worker, start: bool = True, exclusive: bool = True
+    ) -> None:
+        """Add a new worker.
+
+        Args:
+            worker: A Worker instance.
+            start: Start the worker if True, otherwise the worker must be started manually.
+            exclusive: Cancel all workers in the same group as `worker`.
+        """
+        if exclusive and worker.group:
+            self.cancel_group(worker.node, worker.group)
+        self._workers.add(worker)
+        if start:
+            worker._start(self._app, self._remove_worker)
+
+    def _new_worker(
+        self,
+        work: WorkType,
+        node: DOMNode,
+        *,
+        name: str | None = "",
+        group: str = "default",
+        description: str = "",
+        exit_on_error: bool = True,
+        start: bool = True,
+        exclusive: bool = False,
+    ) -> Worker:
+        """Create a worker from a function, coroutine, or awaitable.
+
+        Args:
+            work: A callable, a coroutine, or other awaitable.
+            name: A name to identify the worker.
+            group: The worker group.
+            description: A description of the worker.
+            exit_on_error: Exit the app if the worker raises an error. Set to `False` to suppress exceptions.
+            start: Automatically start the worker.
+            exclusive: Cancel all workers in the same group.
+
+        Returns:
+            A Worker instance.
+        """
+        worker: Worker[Any] = Worker(
+            node,
+            work,
+            name=name or getattr(work, "__name__", "") or "",
```

Comment:
Don't we already get an empty string from `getattr`?

```suggestion
            name=name or getattr(work, "__name__", ""),
```

Reply:
The original fixed a typing issue IIRC. the `getattr` could in theory return anything.

Reply:
Adding `or ""` won't fix typing because `getattr` can _still_ return anything in theory.
The type of `name` is `Any | str` with or without `or ""` at the end.